src/snowflake/snowpark/_internal/telemetry.py

```diff
@@ -142,8 +145,20 @@ def wrap(*args, **kwargs):
             *plan.api_calls,
             {TelemetryField.NAME.value: f"DataFrame.{func.__name__}"},
         ]
-        # The first api call will indicate whether sql simplifier is enabled.
-        api_calls[0]["sql_simplifier_enabled"] = args[0]._session.sql_simplifier_enabled
+        # The first api call will indicate following:
+        # - sql simplifier is enabled.
+        # - height of the query plan
+        # - number of unique duplicate subtrees in the query plan
+        api_calls[0][TelemetryField.SQL_SIMPLIFIER_ENABLED.value] = args[
+            0
+        ]._session.sql_simplifier_enabled
+        try:
+            api_calls[0][TelemetryField.QUERY_PLAN_HEIGHT.value] = plan.plan_height
+            api_calls[0][
+                TelemetryField.QUERY_PLAN_NUM_DUPLICATE_NODES.value
+            ] = plan.num_duplicate_nodes
```

Comment:
Please do a performance test for the client side.

Reply:
A question is how do we query this telemetry?

Reply:
Query below can be used to extract current sql simplifier telemetry. Since we are reporting query plan height and num duplicate nodes within the same telemetry json, both these values can be extracted similarly. 
```
select d, seq, value from (select parse_json(data):api_calls as d from snowscience.snowpark.client_telemetry_snowpark_function where created_on > dateadd(days, -5, current_timestamp) limit 100)  t1, lateral flatten(input => t1.d)
where value like '%sql_simplifier_enabled%';
```

Reply:
> Please do a performance test for the client side.

Let me do a test and report back

Reply:
I did a test run with dummy ctc-repro with 50 repeated self joins the calculation time was about 0.2 seconds for the largest graph
```
sending telemetry: {'name': 'Session.sql', 'sql_simplifier_enabled': True, 'query_plan_height': 2, 'query_plan_num_duplicate_nodes': 0}, elapsed time = 0.0002963542938232422
sending telemetry: {'name': 'Session.table', 'sql_simplifier_enabled': True, 'query_plan_height': 26, 'query_plan_num_duplicate_nodes': 21}, elapsed time = 0.2114887237548828
sending telemetry: {'name': 'Session.sql', 'sql_simplifier_enabled': True, 'query_plan_height': 4, 'query_plan_num_duplicate_nodes': 0}, elapsed time = 0.0007040500640869141
```

Reply:
it seems not bad

Reply:
How about testing 500, 1000 and 5000 logical plans?